test/common/tracing/http_tracer_impl_test.cc

```diff
@@ -266,6 +266,7 @@ TEST(HttpConnManFinalizerImpl, NullRequestHeaders) {
   EXPECT_CALL(request_info, bytesSent()).WillOnce(Return(11));
   Optional<uint32_t> response_code;
   EXPECT_CALL(request_info, responseCode()).WillRepeatedly(ReturnRef(response_code));
+  EXPECT_CALL(request_info, upstreamHost()).WillOnce(Return(nullptr));
```

Comment:
can you also check that there is Times(0) calls to *span, setTag("upstream_cluster"....) in this case.